cristin-import/src/main/java/no/unit/nva/cristin/mapper/artisticproduction/CristinArtisticProduction.java

```diff
@@ -130,4 +164,187 @@ private String[] descriptionFields() {
+            extractIsmn()))
+            .orElseThrow();
+    }
+
+    private Ismn extractIsmn() throws InvalidIsmnException {
+        return StringUtils.isNotBlank(ismn) ? new Ismn(ismn) : null;
+    }
+
+    private Optional<AudioVisualPublication> extractAudioVisualPublication() {
+        if (StringUtils.isNotEmpty(isrc) || hasAudioVisialMedium()) {
+            return Optional.of(createAudioVisualPublication());
+        }
+        return Optional.empty();
+    }
+
+    private boolean hasAudioVisialMedium() {
+        return nonNull(medium);
+    }
+
+    private AudioVisualPublication createAudioVisualPublication() {
+        return attempt(() -> new AudioVisualPublication(extractMediumType(),
+            new UnconfirmedPublisher(publisherName),
+            null,
+            List.of(),
+            constructIsrc())).orElseThrow();
+    }
+
+    private Isrc constructIsrc() {
+        return Optional.ofNullable(isrc).map(this::extractIsrc).orElse(null);
+    }
+
+    private Isrc extractIsrc(String isrc) {
+        return attempt(() -> new Isrc(isrc))
+            .orElseThrow(fail -> new InvalidIsrcException(fail.getException()));
+    }
+
+    private MusicMediaSubtype extractMediumType() {
+        return new MusicMediaSubtype(convertCristinMediumToNvaMediumSubtype());
+    }
+
+    private MusicMediaType convertCristinMediumToNvaMediumSubtype() {
+        return Optional.ofNullable(medium).map(this::convertToNvaMediaSubType).orElse(OTHER);
+    }
+
+    private MusicMediaType convertToNvaMediaSubType(String medium) {
+        if (isCompactDisc(medium)) {
+            return COMPACT_DISC;
+        }
+        if (isVinyl(medium)) {
+            return VINYL;
+        }
+        if (isDigitalFile(medium)) {
+            return DIGITAL_FILE;
+        }
+        if (isStreaming(medium)) {
+            return STREAMING;
+        }
+        return OTHER;
+    }
+
+    @JsonIgnore
+    private boolean isStreaming(String medium) {
+        var pattern = Pattern.compile(
+            ".*((strøm)|(Spotify)|(Stream)|(YouTube)|(vimeo))+.*",
+            Pattern.CASE_INSENSITIVE);
+        var matcher = pattern.matcher(medium);
+        return matcher.matches();
+
+    }
+
+    @JsonIgnore
+    private boolean isDigitalFile(String medium) {
+        var pattern = Pattern.compile(
+            ".*((Digital)|(mp3)|(Lydfil))+.*",
+            Pattern.CASE_INSENSITIVE);
+        var matcher = pattern.matcher(medium);
+        return matcher.matches();
+
+    }
+
+    @JsonIgnore
+    private boolean isVinyl(String medium) {
+        var pattern = Pattern.compile(
+            ".*((Plateinnspilling)|(LP))+.*",
+            Pattern.CASE_INSENSITIVE);
+        var matcher = pattern.matcher(medium);
+        return matcher.matches();
+    }
+
+
+    @JsonIgnore
+    private boolean isCompactDisc(String medium) {
+        var pattern = Pattern.compile(
+            ".*((CD)|(Album))+.*",
+            Pattern.CASE_INSENSITIVE);
```

Comment:
I guess this is where you are referring to regex…I'm not sure that there is a better way…GIGO, which makes life miserable.

You can try (though it's case sensitive, so you might want to uppercase/lowercase):

```
        var input = "hello CD, this is CD";
        var things = Stream.of("CD ", "Album");
        var result = things.anyMatch(input::contains);
```

Bear in mind though that the regex could be better if using `\b`

Reply:
sadly the mapping rules needs to be case insensitive.